### PR TITLE
fix(braintree): pre defined environments

### PR DESCRIPTION
**Pull request by simllll**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] fixes a bug introduced here  https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64094

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## braintree ([<kbd>unpkg</kbd>](https://unpkg.com/browse/braintree@latest/))
was missing the following properties:
1. GraphQL
1. version
1. errorTypes
1. exceptions
1. PaymentInstrumentTypes

<details>
<summary>as well as these 4 other properties...</summary>
<p>WebhookNotification, TestingGateway, UsBankAccountVerification, ValidationErrorCodes</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0fbfebc6ca0804bdb84553cfdb0f47491d063e6f
</p>

File: types/braintree/index.d.ts [hunk 1]
```diff
@@ -135,7 +128,7 @@ declare namespace braintree {
         voidedAt: RangeFieldSearchFn<Date>;
     }) => void;
 
-    export type GatewayConfig = KeyGatewayConfig | ClientGatewayConfig | AccessTokenGatewayConfig | CustomConfig;
+    export type GatewayConfig = KeyGatewayConfig | ClientGatewayConfig | AccessTokenGatewayConfig;
```
Comment (smrubin) [inline types/braintree/index.d.ts:131]: Disclosure that I'm not fully up to date on the braintree lib, but know the CustomConfig was added in recently here. What's the rationale for its removal?

Comment (smrubin) [reply types/braintree/index.d.ts:131]: Ah, nevermind, I see the comment you left on the most recent PR that was merged in. Thanks for adding that.

**Review by smrubin — APPROVED** [review]

File: types/braintree/index.d.ts (hunk 1 above)
Comment (simllll) [reply types/braintree/index.d.ts:131]: :) ah perfect,thanks for the fast approval!

**Comment by simllll** [comment]
Ready to merge

**Pull request merged**